src/Youch.js

```diff
@@ -35,8 +35,11 @@ class Youch {
    * @return {Promise}
    */
   _getFrameSource (frame) {
+    const path = frame.getFileName()
+      // Handle the common setup of Webpack
+      .replace(/dist\/webpack:\//g, "")
```

Comment:
Are there any docs around this pattern? Curious to know 😄

Reply:
Webpack by default generates the paths prefixed with `webpack://` protocol, it does not include the absolute paths where the project was built. This can be controlled with [output.devtoolModuleFilenameTemplate](https://webpack.js.org/configuration/output/#output-devtoolmodulefilenametemplate), but adding absolute paths makes no sense, since it’s irrelevant at the run-time, which may be a completely different setup from where the project was built.

Source map file example:

```ts
{"version":3,"sources":["webpack:///webpack/bootstrap","webpack:///src/index.ts",...
```

Webpack is used mostly for in-browser apps, but it sees plenty of use on Node. The source-map-support package reads the sourcemaps and mutates `Error` object on the fly.

It can’t generate information out of nothing though, and interprets `webpack:///` prefix as a relative path with a silly triple path separator. By the time it’s done with mutating the stack, the filepath is `/project/dir/dist/webpack:/src/index.ts` from the original `webpack:///src/index.ts`.

For our need in Youch we’ve lost since the initial source map, since we technically have no path information. So the usual scenario is to show a “Missing frames” message. But! If the project is being worked on, which is likely, and the project is built with default settings, which is likely, we can, in fact, rely on the relative path and glue the directory of the project with the in-Webpack source path if we delete the `dist/webpack` in the middle.

It’s an unreliable hack, but since we start in a position of a loss, any minor chance of suddenly working is great. The worst case disadvantage is if under a crazy set of circumstances the newly created path matches an existing file, but a wrong one. This is very unlikely and, what’s more, the impact is relatively minor: the developer is confused for a moment.